librz/bin/format/te/te.c

```diff
@@ -18,23 +18,43 @@ ut64 rz_bin_te_get_stripped_delta(struct rz_bin_te_obj_t *bin) {
 	return 0LL;
 }
 
+#define macro_str(s) #s
+#define rz_buf_read_le8_at(a, b, c) \
+	(rz_buf_read_at(a, b, (ut8 *)c, sizeof(ut8)) == 1)
+#define parse_header_value(off, bits, key) \
+	do { \
+		if (!rz_buf_read_le##bits##_at(bin->b, off, &bin->header->key)) { \
+			RZ_LOG_ERROR("Cannot read TE_image_file_header." macro_str(key) "\n"); \
+			return false; \
+		} \
+		off += (bits / 8); \
+	} while (0)
+
 static int rz_bin_te_init_hdr(struct rz_bin_te_obj_t *bin) {
 	if (!bin) {
 		return false;
-	}
-	if (!(bin->header = malloc(sizeof(TE_image_file_header)))) {
-		rz_sys_perror("malloc (header)");
-		return false;
-	}
-	if (rz_buf_read_at(bin->b, 0, (ut8 *)bin->header, sizeof(TE_image_file_header)) == -1) {
-		eprintf("Error: read (header)\n");
+	} else if (!bin->kv) {
```

Comment:
Nice catch, the endianness-dependency! 👍

Reply:
Should help with https://github.com/rizinorg/rizin/issues/297

Reply:
there are no tests on TE format